network-area-diagram/src/main/java/com/powsybl/nad/layout/BasicFixedLayoutFactory.java

```diff
@@ -29,4 +29,15 @@ public Layout create() {
         layout.setFixedNodePositions(fixedPositions);
         return layout;
     }
+
+    public Layout create(LayoutFactory layoutFactory) {
+        if (layoutFactory == null) {
+            return create();
+        } else {
+            Layout layout = layoutFactory.create();
+            layout.setInitialNodePositions(fixedPositions);
+            layout.setNodesWithFixedPosition(fixedPositions.keySet());
```

Comment:
You see here that the current `Layout` API is not very convenient, we should modify it.

Reply:
True! I created an issue for that: https://github.com/powsybl/powsybl-diagram/issues/601